daemon/network.go

```diff
@@ -113,7 +114,12 @@ func (daemon *Daemon) CreateNetwork(name, driver string, ipam network.IPAM, opti
 	}
 
 	nwOptions = append(nwOptions, libnetwork.NetworkOptionIpam(ipam.Driver, "", v4Conf, v6Conf))
-	nwOptions = append(nwOptions, libnetwork.NetworkOptionDriverOpts(options))
+	nwOptions = append(nwOptions, libnetwork.NetworkOptionDriverOpts(driverOptions))
+	if len(networkOptions) > 0 {
+		if _, ok := networkOptions[netlabel.Internal]; ok {
+			nwOptions = append(nwOptions, libnetwork.NetworkOptionInternalNetwork())
```

Comment:
I agree with @calavera it doesn't feel quite right. Should we expose the distinction between network options and driver-specific options end-to-end?